tests/integration/test_startup_scripts_execution_state/test.py

```diff
@@ -46,11 +51,40 @@ def test_startup_execution_state(start_cluster):
         == STATE_SUCCESS
     )
 
-    assert (
-        int(
-            bad.query(
-                "SELECT value FROM system.metrics WHERE metric = 'StartupScriptsExecutionState'"
-            ).strip()
-        )
-        == STATE_FAILURE
+    assert_startup_script_failed()
+    bad.stop_clickhouse()
+    # Set throw_on_error: true for the startup_script
+    bad.replace_in_config(
+        "/etc/clickhouse-server/config.d/bad_script.xml",
+        "<!-- THROW_ON_ERROR PLACEHOLDER -->",
+        "<throw_on_error>true</throw_on_error>",
+    )
+    # TODO: REMOVE
+    assert bad.get_process_pid("clickhouse") is None
+    bad.start_clickhouse(start_wait_sec=120, expected_to_fail=True)
+    # server can't start with errors in startup_script
+    assert bad.get_process_pid("clickhouse") is None
+    assert bad.contains_in_log("Failed to parse startup scripts file")
+    # Logs contains the original error
+    assert bad.contains_in_log(
+        "Unknown table expression identifier 'non_existent_table'"
+    )
+    assert bad.contains_in_log("Cannot finish startup_script successfully")
+
+    bad.replace_in_config(
+        "/etc/clickhouse-server/config.d/bad_script.xml",
+        "<throw_on_error>true</throw_on_error>",
+        "<throw_on_error>false</throw_on_error>",
+    )
+    bad.start_clickhouse()
+    assert bad.get_process_pid("clickhouse") is not None
+
+    # startup script wasn't executed, but the server is up
+    assert_startup_script_failed()
```

Comment:
`nit:` No need to make this a separate function, only used once.

Reply:
the function is called also here https://github.com/ClickHouse/ClickHouse/pull/79732/files#diff-2b51641f630bdc30383e6a48ffd39ebd9112f8a3065a6385adea14f4d7323acdR54